synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -513,6 +513,10 @@ async def _download_url(self, url: str, user: str) -> Dict[str, Any]:
 
                 if b"Content-Type" in headers:
                     media_type = headers[b"Content-Type"][0].decode("ascii")
+                    if media_type.startswith("image/"):
+                        scpos = media_type.find(";")
+                        if scpos > 0:
+                            media_type = media_type[:scpos]
```

Comment:
this looks wrong. should we not preserve any mime-type parameters supplied by the origin server?

Reply:
Are there valid cases for images? Currently it breaks getting thumbnail requirements because they're retrieved from `ContentRepositoryConfig` by MIME type and all those types don't include parameters like charset. It will be preserved for HTML because its MIME type doesn't start with `image/`

Reply:
> Are there valid cases for images?

I don't know, but it doesn't seem crazy that there could be?

>  Currently it breaks getting thumbnail requirements because they're retrieved from ContentRepositoryConfig by MIME type and all those types don't include parameters like charset

ok, but we can also fix that by stripping of the parameters before we index into `ContentRepositoryConfig`, rather than stripping off the parameters before we store them in the database.

Reply:
(which also means we don't have to special-case by `media_type.startswith("image/")`)

Reply:
Okay, I moved it to a more appropriate place.